File: Models/Course.cs
namespace CourseService.Models;

public class Course : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }

    public int? InstructorUserId { get; set; }

    public bool IsPublished { get; set; } = false;

    public decimal Price { get; set; } = 0.0m;

    public ICollection<Lesson> Lessons { get; set; } = new List<Lesson>();
}

File: Frontend/src/features/courses/types/course.ts
import type { Lesson } from "@/features/lessons/types/Lesson";
import type { Money } from "./money";

export interface Course {
  id: string;
  title: string;
  description: string;
  imageUrl?: string | null;
  instructorUserId?: string | null;
  isPublished: boolean;
  price: Money;
  lessons?: Lesson[];
  updatedAtUtc: string;
}
